lib/buffer-search.coffee

```diff
@@ -58,6 +59,17 @@ class BufferSearch
       ResultsMarkerLayersByEditor.set(editor, layer)
     layer
 
+  checkSanity: (findPattern) ->
+    findOptions = new FindOptions(@findOptions.serialize())
+    findOptions.set({findPattern})
+    if regex = findOptions.getFindPatternRegex()
+      try
+        buffer = new TextBuffer('')
+        sane = true
+        buffer.scan regex, ({range}) -> sane = false
+        return sane
+    false
```

Comment:
OK -- I was conservative and tried to match exactly how this is done elsewhere, because I'm not so familiar with the entire project and what's likely to change.